### PR TITLE
Simplify site icon animation on hover.

### LINKED ISSUE #38575: Animated backend site logo. (instead of the W logo seen one can see the site logo.)
### Description

Hovering over the backend Site logo (when the W is not in use) I have noticed it becomes animated.
Like so:
Site-logo-backend-animated

It would be very helpful if the site logo automatically fills out the square it is in and does not animate.
The animation is kinda disturbing.

### Step-by-step reproduction instructions

Add a Site logo, and see if it happens to become animated when hovering over it.
Hmm I am sorry but I am having a difficult time figuring out how the site logo got there in the first place.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

WordPress 5.9.
Gutenberg plugin 12.5.3
Twenty Twenty Two. (I have switched between various themes.)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by jasmussen**
## Description

Fixes #38575. Alternative to #38702. 

This PR smooths out the animation that is added to the site icon when present. The animation helps signify the importance of the element:

after

When no icon is applied, the existing hover style remains:

no icon

When you toggle "reduce motion" in your browser or operating system, no animation is shown.

## Testing Instructions

Add a site icon if you have none already. For example insert the site logo block, add and publish and image, then reload the page. Hover the top left button.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

**Comment by Mamaduka** [comment]
Thank you, Joen.

I personally like this animation, and making it smoother will be a good improvement.

As @gwwar pointed out in my PR, you'll also need to update animation for Site Editor:

https://github.com/WordPress/gutenberg/blob/0bad51e52cd32c75420d9241444d0ea18109e5a4/packages/edit-site/src/components/navigation-sidebar/navigation-toggle/index.js#L59-L65

cc @paaljoachim.

**Comment by jasmussen** [comment]
Thanks for looking, and for the reminder! The code drift of having no shared components here is pretty bad 🙈 — but I think I fixed it. Another look?

**Review by Mamaduka — APPROVED** [review]
This looks good 👍 

Since conversation is still ongoing in the issue, let's wait for the feedback from others as well before we merge this.

**Comment by paaljoachim** [comment]
This is a lot better! Much smoother and just a little more subtile. 
I am still not sure we actually need the animation (as it could be the same as the default W), but that is another matter...
I would say that we are good to go with this. It would be nice with a bit more feedback on the issue before merging.

Before:
Custom-Site-Icon

.
.
......
.
.

After (testing this PR):
custom-site-icon-new-animation

**Review by mtias — APPROVED** [review]

**Comment by jasmussen** [comment]
With three approvals and a green button, and since it's not a ton of code, it's tempting to merge this one. I'll let it sit for tomorrow :)

**Comment by jasmussen** [comment]
Cool, I'll land this one but I'm always happy to follow up!

**Pull request merged**
